File: WebApiSpark/src/main/java/Model/Attachment.java
package Model;


import Util.DatabaseValues;

public class Attachment extends IncidentElement
{
    public Attachment( )
    {
        this (
                "",
                ""
        );
    }

    public Attachment(
            String fileName,
            String fileId
    ) {
        super (
                DatabaseValues.Table.ATTACHMENT,
                new DatabaseValues.Column[]
                {
                        DatabaseValues.Column.FILE_NAME,
                        DatabaseValues.Column.FILE_ID
                }
        );

        updateAttributeValue(
                DatabaseValues.Column.FILE_NAME,
                fileName
        );

        updateAttributeValue(
                DatabaseValues.Column.FILE_ID,
                fileId
        );

    }
}

File: IRSClientAngular/src/app/component/attachment/attachment.ts
import { Config } from '../../util/config.service';
import { IncidentElement } from '../report/incident-element';

export class Attachment implements IncidentElement {
  table: string;
  attributes: AttachmentAttributes;
  searchString: string;

  constructor() {
      this.attributes = new AttachmentAttributes();
      this.table = Config.AttachmentTable;
  }

  setSearchString() {
      this.searchString =
      this.attributes.FILE_NAME
  };
}

export class AttachmentAttributes {
    FILE_NAME: string;
}
